### PR TITLE
Multiprocessing in Python lesson

### PR DESCRIPTION
**Pull request by linamnt**

### CODE REVIEW

File: lessons/python/cpu-multiprocessing/lesson.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+<!-- change visible to true if you want it on the site -->
+<!-- remove any tags listed above that are not relevant -->
+
+ - **Authors**: Lina
+ - **Research field**: Physiology
+ - **Lesson topic**: CPU Multiprocessing in Python
+ - **Lesson content URL**: <https://github.com/UofTCoders/studyGroup/tree/gh-pages/lessons/python/cpu-multiprocessing>
+
+## Resources:
+- Lesson material adapted from [PYMOTW lesson](https://pymotw.com/2/multiprocessing/basics.html)
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/lesson.md]: 3

File: lessons/python/cpu-multiprocessing/mp_tutorial_3_main.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+'''
+ Due to the way the new processes are started, the child process needs to be able to import the script containing the target function. Wrapping the main part of the application in a check for __main__ ensures that it is not run recursively in each child as the module is imported. Another approach is to import the target function from a separate script.
+'''
+import multiprocessing
+from mp_tutorial_2 import worker
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/mp_tutorial_3_main.py]: Doesn't exist.

File: lessons/python/cpu-multiprocessing/joblib-tutorial.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+if __name__ == '__main__':
+    inputs = list(range(10))
+    print('Input   :', inputs)
+
+    builtin_outputs = list(map(f, inputs))
+    print('Built-in:', builtin_outputs)
+
+    #number of threads to use
+    pool_size = cpu_count() * 2
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/joblib-tutorial.py:15]: `cpu_count` is undefined.

File: lessons/python/cpu-multiprocessing/joblib-tutorial.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    print("Number of processes:", pool_size)
+    pool = Pool(processes=pool_size, initializer=start_process)
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/joblib-tutorial.py]: `Pool` is undefined.

File: lessons/python/cpu-multiprocessing/joblib_tutorial_2_multiple_args.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from joblib import Parallel, delayed
+from multiprocessing import cpu_count
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/joblib_tutorial_2_multiple_args.py]: Should this be imported from `joblib` or is it essentially the same?

File: lessons/python/cpu-multiprocessing/joblib_tutorial_2_multiple_varying_args.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+from joblib import Parallel, delayed
+from multiprocessing import cpu_count
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/joblib_tutorial_2_multiple_varying_args.py]: Same question as above.

File: lessons/python/cpu-multiprocessing/dask_tutorial.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+'''
+From: http://distributed.readthedocs.io/en/latest/joblib.html
+'''
+
+
+import distributed.joblib
+# Scikit-learn bundles joblib, so you need to import from
+# `sklearn.externals.joblib` instead of `joblib` directly
+from sklearn.externals.joblib import parallel_backend
```
Comment (QuLogic) [inline lessons/python/cpu-multiprocessing/dask_tutorial.py]: Probably depends on packaging; Fedora/Debian would not allow this.

Comment (linamnt) [reply lessons/python/cpu-multiprocessing/dask_tutorial.py]: hmm, okay, i'll make a disclaimer, just so I understand, it's the parallel_backend that would need to be imported from joblib not sklearn in Fedora/Debian?

Comment (QuLogic) [reply lessons/python/cpu-multiprocessing/dask_tutorial.py]: From a non-embedded `joblib`, like `from joblib import parallel_backend`.

File: lessons/python/cpu-multiprocessing/joblib_tutorial_2_multiple_varying_args.py (hunk 6 above)
Comment (linamnt) [reply lessons/python/cpu-multiprocessing/joblib_tutorial_2_multiple_varying_args.py]: does the same, but not sure if actually using same as multiprocessing :/ i'll just import from joblib

**Review by QuLogic — APPROVED** [review]

**Pull request merged**